### PR TITLE
Update the Documentation

### PR DESCRIPTION
**Pull request by envoyr**
New features are the hint in the footer that you can edit this page and will be redirected to the repository.

Furthermore, instructions were updated and marked as tested with the corresponding version.

Since there were outdated texts during the manual installation, they were revised again. There is still a lot of room for improvement, but this avoids using old techniques (libnss-mysql-bg, nscd, php-5) that are no longer actively used by Froxlor.

### CODE REVIEW

**Review by d00p — COMMENTED** [review]
Generally, the setting `PHP-FPM -> Alias folder of php-fpm configuration` is _not_ the directory for the pool-configs!!! This can be set in `admin -> PHP -> PHP-FPM versions`

File: configuration/apache2/fcgid.md [hunk 1]
```diff
@@ -20,214 +20,131 @@ Running PHP as a CGI would require reloading the PHP binary for every request wh
...
-
-Force the cronjob once to run to create the fcgi settings for all domains
-
-    /usr/bin/php5 /var/www/froxlor/scripts/froxlor_master_cronjob.php --force --debug
-
-Don't forget to start Cron again.
-
-    /etc/init.d/cron start
+````shell
+/etc/init.d/cron start
```
Comment (envoyr) [inline configuration/apache2/fcgid.md:140]:
```suggestion
service cron start
```

see `cron` comment above

File: configuration/apache2/php-fpm.md [hunk 2]
```diff
@@ -70,130 +83,74 @@ Done!
...
+You should ensure that the Froxlor cronjob isn't executed while you set up php-fpm. This could produce unwanted results!
 
-You must now instruct Froxlor to create the FPM configuration and to add related directives to Apache's VirtualHost configurations.
-
-In order to use FPM you need to create a user _froxlorlocal_ and a group of the same name:
-
-    addgroup --gid 9999 froxlorlocal
-    adduser --no-create-home --uid 9999 --ingroup froxlorlocal --shell /bin/false --disabled-password --gecos '' froxlorlocal
+````shell
+/etc/init.d/cron stop
```
Comment (envoyr) [inline configuration/apache2/php-fpm.md:100]:
```suggestion
service cron stop
```

see `cron` comment above

File: configuration/apache2/php-fpm.md [hunk 3]
```diff
@@ -70,130 +83,74 @@ Done!
...
-    apt-get install apache2-mpm-worker
-
-You also want to restart the Cron daemon now:
+ ````shell
+php /var/www/froxlor/scripts/froxlor_master_cronjob.php --force
+````
 
-    /etc/init.d/cron start
+````shell
+/etc/init.d/cron start
```
Comment (envoyr) [inline configuration/apache2/php-fpm.md:138]:
```suggestion
service cron start
```

see `cron` comment above

File: configuration/nginx/php-fpm.md [hunk 4]
```diff
@@ -70,110 +87,66 @@ Done!
...
+Run Froxlor's global cron job once to immediately produce php-fpm configurations for all VirtualHosts:
 
-Start the system cron demon again.
+ ````shell
+php /var/www/froxlor/scripts/froxlor_master_cronjob.php --force
+````
 
-    /etc/init.d/cron start
+````shell
+/etc/init.d/cron start
```
Comment (envoyr) [inline configuration/nginx/php-fpm.md:142]:
```suggestion
service cron start
```

see `cron` comment above

**Review by envoyr — COMMENTED** [review]
Sorry for the late reply, i was a little bit busy. I have added some suggestions for improvement.

File: configuration/apache2/fcgid.md [hunk 5]
```diff
@@ -20,214 +20,131 @@ Running PHP as a CGI would require reloading the PHP binary for every request wh
...
+### 3.1. Quick installation
 
-Now restart Apache with
+This method is especially useful for freshly installed Froxlor instances.
 
-    /etc/init.d/apache2 restart
+#### 3.1.1. Setting up the environment
 
-If you don't see any new errors or warnings than without suexec you can continue with the next section. Otherwise check the logs, they will help you.
+In order to use fcgid in Froxlor you need to create a `froxlorlocal` *user* and *group* of the same name:
```
Comment (d00p) [inline configuration/apache2/fcgid.md]: The config-template do that according to the settings. The name of the local user can be changed in the settings

File: configuration/apache2/fcgid.md [hunk 6]
```diff
@@ -20,214 +20,131 @@ Running PHP as a CGI would require reloading the PHP binary for every request wh
...
 
-Now restart Apache with
+#### 3.2.1. Setting up the environment
 
-    /etc/init.d/apache2 restart
+In order to use FCGID in Froxlor you need to create a `froxlorlocal` *user* and *group* of the same name:
 
-Now follow the next steps and customize the fcgi path in Froxlor
+```shell
+addgroup --gid 9999 froxlorlocal
```
[the remark this answers is not in the document]
Comment (d00p) [reply configuration/apache2/fcgid.md:86]: well config-services script does the exact same commands as the configuration-templates show when doing it manually. So,adding the user is still included there

File: configuration/apache2/php-fpm.md [hunk 7]
```diff
@@ -70,130 +74,74 @@ Done!
...
-
-If you don't see a result like this or just the virtual user IDs, re-check your configuration and whether these users really exist using the **getent passwd** command. If nscd isn't working properly you cannot continue with the next steps.
-
-#### 3.2.3. Setting up fastcgi
-
-Now edit /etc/apache2/mods-available/fastcgi.conf - it needs to look like this:
+In order to use FPM in Froxlor you need to create a `froxlorlocal` *user* and *group* of the same name:
 
+```shell
+addgroup --gid 9999 froxlorlocal
```
Comment (d00p) [inline configuration/apache2/php-fpm.md:84]: same as for the other one - not needed as it is included in the config-templates

File: configuration/nginx/php-fpm.md [hunk 8]
```diff
@@ -55,13 +61,24 @@ System settings
...
 
-#### 3.1.1 Run auto installer
+This method is especially useful for freshly installed Froxlor instances.
+
+#### 3.1.1. Setting up the environment
+
+In order to use FPM in Froxlor you need to create a `froxlorlocal` *user* and *group* of the same name:
+
+```shell
+addgroup --gid 9999 froxlorlocal
```
Comment (d00p) [inline configuration/nginx/php-fpm.md]: also not required, part of config-templates

File: configuration/nginx/php-fpm.md [hunk 9]
```diff
@@ -70,110 +87,66 @@ Done!
...
-This should produce a result like:
-
-    drwxr-xr-x  3 web1      web1       4096 2008-06-13 17:18 web1
-    drwxr-xr-x  8 web2      web2       4096 2009-01-02 17:40 web2
-    drwxr-xr-x 10 web3      web3       4096 2009-01-25 13:33 web3
+In order to use FPM in Froxlor you need to create a `froxlorlocal` *user* and *group* of the same name:
 
-If you don't see a result like this or just the virtual user IDs, re-check your configuration and whether these users really exist using the **getent passwd** command. If nscd isn't working properly you cannot continue with the next steps.
+```shell
+addgroup --gid 9999 froxlorlocal
```
Comment (d00p) [inline configuration/nginx/php-fpm.md:88]: and again :)

File: installation/debian.md [hunk 10]
```diff
@@ -30,7 +30,31 @@ Install Froxlor together with all needed packages (e.g. webserver):
...
+````
+mysql -u root
+````
+
+To create the users, we execute the following commands, **please change the default passwords**:
+
+````
+ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY 'CHANGEM3';
+CREATE USER 'froxlor'@'localhost' IDENTIFIED BY 'CHANGEM3';
+GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, INDEX, DROP, ALTER, CREATE TEMPORARY TABLES, LOCK TABLES ON froxlor.* TO 'froxlor'@'localhost';
```
Comment (d00p) [inline installation/debian.md:52]: why not just use `GRANT ALL PRIVILEGES...`?

**Review by d00p — CHANGES_REQUESTED** [review]
still adding local user is in there but it's based on a setting and is part of the config-templates

**Comment by envoyr** [comment]
Sorry I messed up the pull request a bit, I'm not that familiar with the reviews. Had given you a wrong version to review. 🙈 

> Generally, the setting `PHP-FPM -> Alias folder of php-fpm configuration` is _not_ the directory for the pool-configs!!! This can be set in `admin -> PHP -> PHP-FPM versions`

Should this info be deleted from the documentation?

**Comment by d00p** [comment]
> Should this info be deleted from the documentation?

it's just important to note that the configuration path for the pool-files (domain-based settings) can be configured in `admin -> PHP -> PHP-FPM versions` and that `PHP-FPM -> Alias folder of php-fpm configuration` is just the wrong setting for that :)

File: configuration/apache2/fcgid.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (envoyr) [reply configuration/apache2/fcgid.md:86]: Ahh i understand, didn't see that before. 👍

**Review by envoyr — COMMENTED** [review]
Updated in commit ed7f42a

**Pull request merged**